ui/store/controller-actions/network-order-controller.ts

```diff
@@ -22,20 +27,35 @@ export function enableAllPopularNetworks(): ThunkAction<
 > {
   return async (dispatch, getState) => {
     const state = getState();
-    const [caipNetworkConfigMap] =
+    const [multichainNetworkConfigMap] =
       getMultichainNetworkConfigurationsByChainId(state);
+    const isMultichainState2Enabled =
+      getIsMultichainAccountsState2Enabled(state);
+
+    // Choose the appropriate featured networks constant based on feature flag
+    const featuredNetworkChainIds = isMultichainState2Enabled
+      ? FEATURED_NETWORK_CHAIN_IDS_MULTICHAIN
+      : FEATURED_NETWORK_CHAIN_IDS;
 
-    const evmPopularNetworkChainIds = FEATURED_NETWORK_CHAIN_IDS.filter(
-      (chainId) => toEvmCaipChainId(chainId) in caipNetworkConfigMap,
-    );
-    await dispatch(
-      setEnabledNetworks(evmPopularNetworkChainIds, KnownCaipNamespace.Eip155),
-    );
+    // Filter to get only available EVM networks
+    const availableEvmNetworks = featuredNetworkChainIds
+      .filter((chainId): chainId is Hex => isHexString(chainId as Hex))
+      .filter(
+        (chainId) => toEvmCaipChainId(chainId) in multichainNetworkConfigMap,
+      );
+
+    // Enable all available EVM networks in a single call
+    if (availableEvmNetworks.length > 0) {
+      await dispatch(
+        setEnabledNetworks(availableEvmNetworks, KnownCaipNamespace.Eip155),
+      );
+    }
   };
 }
 
 export function enableSingleNetwork(
   chainId: Hex | CaipChainId,
+  isMultichainAccountsFeatureEnabled: boolean = false,
```

Comment:
Lets move this inside the function instead of a parameter. This will keep the interface clean and reduce files touched.